jvb/src/test/kotlin/org/jitsi/videobridge/cc/allocation/BitrateControllerTest2.kt

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright @ 2021 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.videobridge.cc.allocation
+
+import io.kotest.core.spec.IsolationMode
+import io.kotest.core.spec.style.ShouldSpec
+import io.kotest.matchers.ints.shouldBeLessThan
+import org.jitsi.nlj.MediaSourceDesc
+import org.jitsi.nlj.RtpEncodingDesc
+import org.jitsi.nlj.util.bps
+import org.jitsi.test.time.FakeClock
+import org.jitsi.videobridge.util.VideoType
+import java.io.File
+import java.time.Instant
+
+/**
+ * This test reads a trace file containing bandwidth estimations and bitrates for individual layers captured from a
+ * "real" conference. The conference had a total of 7 endpoints (one receiver and 6 senders), and the bridge->receiver
+ * BWE was artificially reset to 0 just prior to taking the trace, simulating a ramp-up from 0.
+ *
+ * While [BitrateControllerTest] tests the individual decisions made by [BandwidthAllocator] when the bitrates of the
+ * layers are fixed, here we test for the number of allocation changes which are visible to the user as changes in the
+ * number and/or quality of the videos being shown. Specifically, we want to limit the "flickering" which happens as
+ * a result of fluctuations in the bitrates of the layers and the changes in BWE.
+ */
+class BitrateControllerTest2 : ShouldSpec() {
+    override fun isolationMode() = IsolationMode.InstancePerLeaf
+
+    private val clock = FakeClock()
+    private val A = Endpoint("A")
+    private val B = Endpoint("B")
+    private val C = Endpoint("C")
+    private val D = Endpoint("D")
+    private val E = Endpoint("E")
+    private val F = Endpoint("F")
+    private val bc = BitrateControllerWrapper(listOf(A, B, C, D, E, F), clock = clock)
+
+    init {
+        bc.bc.endpointOrderingChanged()
+        val parsedLines = File("${System.getProperty("user.dir")}/src/test/resources/bwe-events.csv")
```

Comment:
Is this guaranteed to be right both for mvn tests and IntelliJ?

Reply:
> Is this guaranteed to be right both for mvn tests and IntelliJ?

In my environment it works in both

Reply:
BitrateControllerTest2::class.java.getResourceAsStream("/bwe-events.csv") would be the proper way to retrieve a resource file.